draft-cpbs-pseudorandom-ctls.md

```diff
@@ -112,14 +110,25 @@ Pseudorandom cTLS also enciphers every record header.  In addition to the header
 
 By default, Pseudorandom cTLS assumes that the TLS ciphertext is using an AEAD algorithm whose output is purely pseudorandom, such as AES-GCM and ChaCha20-Poly1305.  If the ciphertext might not be pseudorandom (e.g. when handling hostile plaintext), the ciphertext can be "reciphered" to ensure pseudorandomness (see {{confusion-defense}}).
 
+### Key Derivation
+
+To provide clear separation between data sent by the client and the server, the client and server encipher data using different keys, derived from the profile key as follows:
+
+~~~
+client_key = TSPRP-Encipher(key, "derive", zeros)
+server_key = TSPRP-Encipher(key, "derive", ones)
```

Comment:
It may wind up being more efficient to fix the plaintext input to, say, the empty string, and vary the tweak.

Reply:
I thought of that initially, but I wrote it this way because

1. This absolutely guarantees that `client_key` and `server_key` are distinct (due to invertibility).
2. This seems to guarantee that there is no related-key attack between them, because any such attack would violate the SPRP definition.

I've never been entirely clear on what the guarantees are about the (un)relatedness of differently-tweaked instances of the TSPRP, so that seemed harder to reason about.

Reply:
For a secure TSPRP: For each key K and distinct tweaks T, T', TSPRP-Encipher(K, T, .) and TSPRP-Encipher(K, T', .) can be treated as independent, random random permutations. See Section 2 of [Lis02](https://link.springer.com/content/pdf/10.1007/3-540-45708-9_3.pdf).

Reply:
Note that the input can't be "the empty string".  It would have to be `zeros`, the length of the key.  I'm pretty sure this is strictly more efficient than anything involving two different tweaks.

Reply:
Indeed you're right. This is fine.